snap/validate.go

```diff
@@ -81,6 +82,99 @@ func ValidateAlias(alias string) error {
 	return nil
 }
 
+// ValidateAppSocketName checks if a string ca be used as a name for a socket
+// (for socket activation).
+func ValidateAppSocketName(name string) error {
+	valid := validSnapName.MatchString(name)
+	if !valid {
+		return fmt.Errorf("invalid app socket name: %q", name)
+	}
+	return nil
+}
+
+// ValidateAppSocketListenAddress checks that the value of socket addresses.
+func ValidateAppSocketListenAddress(socket *SocketInfo, fieldName string, address string) error {
+	if address == "" {
+		return fmt.Errorf("socket %q must define %q", socket.Name, fieldName)
+	}
+
+	switch address[0] {
+	case '/', '$':
+		return validateAppSocketListenAddressPath(socket, fieldName, address)
+	case '@':
+		return validateAppSocketListenAddressAbstractSocket(socket, fieldName, address)
+	default:
+		return validateAppSocketListenAddressNetAddress(socket, fieldName, address)
+	}
+}
+
+func validateAppSocketListenAddressPath(socket *SocketInfo, fieldName string, path string) error {
+	if path[0] == '$' {
+		if strings.HasPrefix(path, "$SNAP_DATA/") || strings.HasPrefix(path, "$SNAP_COMMON/") {
+			return nil
+		}
+
+		return fmt.Errorf(
+			"socket %q has invalid %q: only $SNAP_DATA and $SNAP_COMMON prefixes are allowed", socket.Name, fieldName)
+	}
+
+	if path[0] == '/' {
```

Comment:
We really want the variables here instead of the absolute paths, since we're introducing this just now and there's apparently no good reason to hardcode those paths.